lib/galaxy/schema/schema.py

```diff
@@ -1390,21 +1410,29 @@ class HistoryDetailed(HistorySummary):  # Equivalent to 'dev-detailed' view, whi
     )
 
 
+class HistoryDevDetailed(HistoryCommonDetailed):
+    """View used by the client to display "some" detailed history information.
+
+    Currently used by the `HistoryController` endpoints.
+    """
+
+    contents_active: HistoryActiveContentCounts = Field(
```

Comment:
Could you request this separately instead, like the history grid does ? (e.g. https://usegalaxy.org/api/histories/18173ac7e98c6e7a?keys=nice_size,contents_active,contents_states) ?

Reply:
This model is only used by the `history/current_history_json` (not the regular API) and I'm using it right now just to know what fields we dealing with in the store.

Reply:
Sounds good, but let's not add it to the AnyHistoryView then ?

Reply:
That is a good point, I should have mentioned that, unfortunately, if the model is not referenced in any API route, then, the client schema, does not include/generate it :disappointed: 

In theory, this dev-detailed model is a "valid" view that we can request with `view=dev-detailed` https://github.com/galaxyproject/galaxy/blob/6b96bb17b2099c527fd5e6b02699e8f6b81b6593/lib/galaxy/managers/histories.py#L816 so doesn't seem to hurt to put it in the AnyHistoryView for now.

We should work on `dev` on migrating this `history/current_history_json` to a regular API endpoint that also supports other histories apart from the current one and maybe that would be the perfect time to fine-tune which fields we should really serialize and then request the expensive ones on demand from the client. All this with proper strong types both in the backend and frontend.

Reply:
Is there maybe a lighter-weight way to handle this ? I dislike adding types to our API that aren't actually part of our API ... could you take `CustomHistoryView` and pick the fields in your type declaration ?

Reply:
Yes I could just declare the interface in the client and drop it from the "AnyHistoryView" it is something temporal anyway :+1: